File: Java/콜라츠 추측.java
// 콜라츠 추측
public class Q17 {
	public static void main(String[] args) {
		int n = 626331;
		Guess guess = new Guess();
		System.out.println(guess.guess(n));
	}
}
class Guess {
    public int guess(long num) {
    	int count = 0;
        while(num>1) {
        	count++;
        	if(count>=500) {
        		count=-1;
        		break;
        	}
        	if(num%2==0) {
        		num/=2;
        	}
        	else {
        		num=(num*3)+1;
        	}
        }
        return count;
    }
}

File: JavaScript/콜라츠 추측.js
// 프로그래머스 Lv 1
// 콜라츠 추측

const solution = (n) => {
  let count = 0;
  while (n > 1 && count < 500) {
    n = n % 2 === 0 ? n / 2 : n * 3 + 1;
    count++;
  }
  return count === 500 ? -1 : count;
};

const n = 1;
const n2 = 16;
const n3 = 626331;

console.log(solution(n));
console.log(solution(n2));
console.log(solution(n3));
